src/libmeasurement_kit/dns/system_resolver_impl.hpp

```diff
@@ -1,194 +0,0 @@
-
-#include <cassert>
-#include <future>
-
-namespace mk {
-namespace dns {
-
-class ResolverContext {
-  public:
-    QueryClass dns_class;
-    QueryType dns_type;
-    std::string name;
-    Callback<Error, Var<Message>> cb;
-    Settings settings;
-    Var<Reactor> reactor;
-    Var<Logger> logger;
-
-    addrinfo hints;
-
-    Var<Message> message{new Message};
-
-    ResolverContext(QueryClass dns_c, QueryType dns_t, std::string n,
-                    Callback<Error, Var<Message>> c, Settings s, Var<Reactor> r,
-                    Var<Logger> l) {
-        dns_class = dns_c;
-        dns_type = dns_t;
-        name = n;
-        cb = c;
-        settings = s;
-        reactor = r;
-        logger = l;
-        memset(&hints, 0, sizeof (hints));
-    }
-};
-
-class addrinfo_deleter {
-  public:
-    void operator()(addrinfo *ptr) {
-        if (ptr) {
-            freeaddrinfo(ptr);
-        }
-    }
-};
-using addrinfo_uptr = std::unique_ptr<addrinfo, addrinfo_deleter>;
-
-template <MK_MOCK(getaddrinfo), MK_MOCK(inet_ntop)>
-void resolve_async_impl(ResolverContext *context) {
-    std::unique_ptr<ResolverContext> ctx(context);
-    Callback<Error, Var<Message>> callback = ctx->cb;
-    Var<Message> message = ctx->message;
-    addrinfo *sip = nullptr;
-
-    int error = getaddrinfo(ctx->name.c_str(), nullptr, &ctx->hints, &sip);
-    ctx->logger->log(MK_LOG_DEBUG2, "getaddrinfo result: %d", error);
-    if (error) {
-        Error resolver_error;
-        switch (error) {
-        case EAI_AGAIN:
-            resolver_error = TemporaryFailureError();
-            break;
-        case EAI_BADFLAGS:
-            resolver_error = InvalidFlagsValueError();
-            break;
-#ifdef EAI_BADHINTS  // Not always available
-        case EAI_BADHINTS:
-            resolver_error = InvalidHintsValueError();
-            break;
-#endif
-        case EAI_FAIL:
-            resolver_error = NonRecoverableFailureError();
-            break;
-        case EAI_FAMILY:
-            resolver_error = NotSupportedAIFamilyError();
-            break;
-        case EAI_MEMORY:
-            resolver_error = MemoryAllocationFailureError();
-            break;
-        case EAI_NONAME:
-            resolver_error = HostOrServiceNotProvidedOrNotKnownError();
-            break;
-        case EAI_OVERFLOW:
-            resolver_error = ArgumentBufferOverflowError();
-            break;
-#ifdef EAI_PROTOCOL  // Not always available
-        case EAI_PROTOCOL:
-            resolver_error = UnknownResolvedProtocolError();
-            break;
-#endif
-        case EAI_SERVICE:
-            resolver_error = NotSupportedServnameError();
-            break;
-        case EAI_SOCKTYPE:
-            resolver_error = NotSupportedAISocktypeError();
-            break;
-        default:
-            resolver_error = ResolverError();
-            break;
-        }
-        ctx->reactor->call_soon([=]() { callback(resolver_error, nullptr); });
-        return;
-    }
```

Comment:
This switch is now a function in `getaddrinfo_async.hpp` that just maps the error.